teps/0102-https-connection-to-triggers-interceptor.md

```diff
@@ -0,0 +1,82 @@
+---
+title: HTTPS Connection to Triggers ClusterInterceptor
+authors:
+  - "@savitaashture"
+creation-date: 2022-03-21
+last-updated: 2022-03-21
+status: proposed
+---
+
+# TEP-0102: HTTPS Connection to Triggers ClusterInterceptor
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+- [Design Details](#design-details)
+- [Implementation Details](#implementation-details)
+- [A look into the future](#a-look-into-the-future)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes to run ClusterInterceptor server as HTTPS and to provide a secure connection between 
+EventListener and ClusterInterceptor. 
+
+## Motivation
+
+Running ClusterInterceptor server as HTTPS provide a secure connection to eventlistener
+and also helps triggers users where there is a hard requirement to handle all connections securely.
+
+### Goals
+
+* Running ClusterInterceptor as `HTTPS` and making sure a secure connection between Eventlistener and ClusterInterceptor.
+* No configuration changes asked from user.
+
+### Non-Goals
+
+* Requiring inputs from end user
+
+## Proposal
+
+Triggers now have full support of end to end secure connection by Running ClusterInterceptor as `HTTPS`.
+
+### User Stories
+
+* ClusterInterceptor calls are done using `HTTP` instead of `HTTPS` which is considered a security problem because
+in many environments(like OpenShift Container Platform 4) there is a hard requirement to have all traffic using `HTTPS`. 
+Meaning all traffic needs to be secured. Since ClusterInterceptor are not offering HTTPS, they can't be used unless they are offering HTTPS.
+Hence the goal is to make sure that all ClusterInterceptor calls are using HTTPS instead, to comply with security regulation/requirements.
+
+## Design Details
+
+* By default ClusterInterceptor run as `HTTPS`.
+* There won't be a support for `HTTP`.
+
+    Reason:
+    1. Support of `HTTPS` doesn't require any user intervention and providing secure connection is more preferred than insecure connection.
+    2. If we plan to support both `HTTP` and `HTTPS` we require input from user as well as need to add condition checks to execute based on user provided input. 
```

Comment:
@dibyom added reason for not supporting `HTTP` connection

Please do let me know if require to add more info